### PR TITLE
Image Customizer: Handle removal of mkinitrd in 3.0.

### PR DESCRIPTION
**Pull request by cwize1**
In Azure Linux 3.0, the `mkinitrd` script was removed in favor of using `dracut` directly. This change fixes the image customizer tool so that it uses the appropriate tool for updating the initramfs file depending on the base image being used.

---

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [x] Ready to merge

---

###### Does this affect the toolchain?  

NO

###### Test Methodology

- Manually ran image customizer tool.

### CODE REVIEW

**Review by romoh — APPROVED** [review]

**Review by gmileka — APPROVED** [review]

File: toolkit/tools/internal/file/file.go [hunk 1]
```diff
@@ -429,3 +430,14 @@ func EnumerateDirFiles(dirPath string) (filePaths []string, err error) {
 	}
 	return filePaths, nil
 }
+
+func CommandExists(name string) (bool, error) {
+	_, err := exec.LookPath("mkinitrd")
```
Comment (PawelWMS) [inline toolkit/tools/internal/file/file.go:435]:
```suggestion
	_, err := exec.LookPath(name)
```

File: toolkit/tools/internal/file/file.go [hunk 2]
```diff
@@ -429,3 +430,14 @@ func EnumerateDirFiles(dirPath string) (filePaths []string, err error) {
 	}
 	return filePaths, nil
 }
+
+func CommandExists(name string) (bool, error) {
```
Comment (PawelWMS) [inline toolkit/tools/internal/file/file.go:434]: Could you also add tests for the new function?

Comment (PawelWMS) [reply toolkit/tools/internal/file/file.go:434]: I'm guessing all systems will have at least `ls` or other similar command we can test against.

File: toolkit/tools/internal/file/file.go (hunk 1 above)
Comment (cwize1) [reply toolkit/tools/internal/file/file.go]: Thanks! Fixed.

File: toolkit/tools/internal/file/file.go (hunk 2 above)
Comment (cwize1) [reply toolkit/tools/internal/file/file.go:434]: Fixed.

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**